packages/shorebird_cli/lib/src/shorebird_flavor_mixin.dart

```diff
@@ -0,0 +1,120 @@
+import 'dart:io';
+
+import 'package:collection/collection.dart';
+import 'package:path/path.dart' as p;
+import 'package:platform/platform.dart';
+import 'package:shorebird_cli/src/command.dart';
+
+/// Mixin on [ShorebirdCommand] which exposes methods for extracting
+/// product flavors from the current app.
+mixin ShorebirdFlavorMixin on ShorebirdCommand {
+  /// Return the set of product flavors configured for the app at [path].
+  /// Returns an empty set for apps that do not use product flavors.
+  Future<Set<String>> extractProductFlavors(
+    String path, {
+    Platform platform = const LocalPlatform(),
+  }) async {
+    final executable = platform.isWindows ? 'gradlew.bat' : 'gradlew';
```

Comment:
I think we are going to want to pull this out into some sort of class at some point (not necessarily now).  We will have a lot of these kinds of binaries we switch on windows and also many we need to find/configure the paths for.

Reply:
yeah I thought about introducing a new class for this but decided we can do that when we need to use this in two places 🤷